src/libcore/cell.rs

```diff
@@ -235,7 +236,8 @@ use ptr;
 ///
 /// See the [module-level documentation](index.html) for more.
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct Cell<T> {
+#[repr(transparent)]
+pub struct Cell<T: ?Sized> {
     value: UnsafeCell<T>,
```

Comment:
Ping @eddyb is `UnsafeCell` itself already effectively `repr(transparent)`?

Reply:
It should be explicitly marked as such.

Reply:
Note that `#[repr(transparent)]` means the same thing as `#[repr(C)]` except it also guarantees you can *pass it by value* and get the same *call ABI* as its only non-ZST field. If we don't want to make the call ABI transparency guarantee, we should use only `#[repr(C)]`. Although, transparency would be nice. So maybe we should have a proper decision on it?

Reply:
Also, in terms of implementation details, within the libcore/libstd you can "just" assume that anything newtype-shaped already has the memory layout and call ABI of the inner type. So this PR *by itself* needs no annotations, and adding annotations should be considered as sort of insta-stable "guarantees".

Reply:
> It should be explicitly marked as such.

Should that be a separate PR or happen here?

Just trying to make sure it doesn't get forgotten.

Reply:
cc @rust-lang/lang @rust-lang/libs Whose responsibility is it, for the addition of insta-stable `#[repr]` attributes like these ones?

Reply:
For what it’s worth we’re about to stabilize the attribute itself: https://github.com/rust-lang/rust/issues/43036

Reply:
The attribute is stable. https://github.com/rust-lang/rust/pull/51395 adds `#[repr(transparent)]` to `UnsafeCell`. (It’s waiting on checkboxes from team members.)